libs/market-list/src/lib/utils/market-list.utils.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  MarketList,
+  MarketList_markets,
+  MarketList_markets_candles,
+} from '../components/__generated__/MarketList';
+
+export const priceChange = (m: MarketList_markets) => {
+  if (
+    m.candles &&
+    m.candles.length > 0 &&
+    m.candles[m.candles.length - 1]?.open &&
+    m.candles[m.candles.length - 1]?.close
+  ) {
+    const yesterdayLastPrice = Number(m.candles[0]?.close);
+    const recentLastPrice = Number(m.candles[m.candles.length - 1]?.close);
+    const increase = recentLastPrice - yesterdayLastPrice;
+
+    return (increase / yesterdayLastPrice) * 100;
+  }
+  return 0;
+};
+
+export const candles = (m: MarketList_markets) =>
+  m.candles?.map((c: MarketList_markets_candles | null) => ({
+    open: price(c?.open, m.decimalPlaces),
+    close: price(c?.close, m.decimalPlaces),
+  }));
+
+export const price = (
+  value: number | string | null | undefined,
+  decimalPlaces: number
+) =>
+  value === undefined || value === null
+    ? 'N/A'
+    : Number(value) / Math.pow(10, decimalPlaces);
+
+export const marketCode = (m: MarketList_markets) =>
+  m.tradableInstrument.instrument?.code;
+
+export const lastPrice = (m: MarketList_markets) =>
+  m.candles
```

Comment:
Chore (blocking): Given you need `m.candles.length` on 43, I think a better predicate here would be `m.candles?.length > 0`. This removes any need for the elvis operators on line 43 (where you're also checking for `m.candles`, so that can be removed)